pdf/creator/creator_test.go

```diff
@@ -3110,32 +3095,76 @@ func testWriteAndRender(t *testing.T, c *Creator, pname string) {
 	testRender(t, pname)
 }
 
-func testRender(t *testing.T, pname string) {
-	tname := strings.TrimSuffix(filepath.Base(pname), filepath.Ext(pname))
+func testRender(t *testing.T, pdfPath string) {
+	if baselineRenderPath == "" {
+		t.Skip("skipping render tests; set UNIDOC_RENDERTEST_BASELINE_PATH to run")
+	}
+	// Set to true to create the baseline.
+	saveBaseline := false
+
+	// Write rendering outputs to a temporary directory that is cleaned up afterwards.
+	tempDir, err := ioutil.TempDir("", "unidoc")
+	if err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	tplName := strings.TrimSuffix(filepath.Base(pdfPath), filepath.Ext(pdfPath))
 	t.Run("render", func(t *testing.T) {
-		if baselineRenderPath == "" {
-			t.Skip("skipping render tests; set UNIDOC_RENDERTEST_BASELINE_PATH to run")
-		}
-		fname := filepath.Join(baselineRenderPath, tname)
-		// will emit template_1-x.png
-		err := renderPDFToPNGs(pname, 0, fname)
+		imgPathPrefix := filepath.Join(tempDir, tplName)
+		imgPathTpl := filepath.Join(tempDir, tplName) + "-%d.png"
+		// will emit /tmp/dir/template-x.png for each page x.
+		err := renderPDFToPNGs(pdfPath, 0, imgPathTpl)
 		if err != nil {
 			t.Skip(err)
 		}
 		for i := 1; true; i++ {
-			name1 := fmt.Sprintf(fname+"-%d.png", i)
-			name2 := fmt.Sprintf(fname+"-%d_exp.png", i)
-			if _, err := os.Stat(name1); i > 1 && err != nil {
+			imgPath := imgPathPrefix + fmt.Sprintf("-%d.png", i)
```

Comment:
Might be clearer to add everything to Sprintf since it's already used:
```
imgPath := fmt.Sprintf("%s-%d.png", imgPathPrefix, i)
expImgPath := filepath.Join(baselineRenderPath, fmt.Sprintf("%s-%d_exp.png", tplName, i))
```